lib/game.rb

```diff
@@ -0,0 +1,102 @@
+require_relative './player.rb'
+require_relative './board.rb'
+
+class Game
+	attr_accessor :board, :player_1, :player_2, :current_player
+
+	WIN_COMBINATIONS = [
+	[0, 1, 2],
+	[3, 4, 5],
+	[6, 7, 8],
+	[0, 3, 6],
+	[1, 4, 7],
+	[2, 5, 8],
+	[0, 4, 8],
+	[2, 4, 6]
+	]
+
+	@@wins = 0
+	@@draws = 0
+	def initialize(player_1=Players::Human.new("X"), player_2=Players::Human.new("O"), board=Board.new)
+		@player_1 = player_1
+		@player_2 = player_2
+		@board = board
+		@current_player = player_1
+		@winning_symbol
+	end
+
+	def over? 
+		self.won? || self.draw? ? true : false
+	end
+
+	def won?
+		WIN_COMBINATIONS.each do |combo|
+			board = self.board.cells
+			if board[combo[0]] == "X" && board[combo[1]] == "X" && board[combo[2]] == "X" ||
+			   board[combo[0]] == "O" && board[combo[1]] == "O" && board[combo[2]] == "O"
+			   @winning_symbol = board[combo[0]]
+			   return true
+			end
+		end
+		false
+	end
+
+	def draw?
+		!self.board.cells.include?(" ") && !self.won? ? true : false
+	end
+
+	def winner
+		if self.won?
```

Comment:
Another alternative to solving this is

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.